cocotb/utils.py

```diff
@@ -545,3 +545,123 @@ def remove_traceback_frames(tb_or_exc, frame_names):
             assert tb.tb_frame.f_code.co_name == frame_name
             tb = tb.tb_next
         return tb
+
+
+class SimTime():
+
+    def __init__(self, time, unit=None):
+        if isinstance(time, SimTime):
+            tme = time.steps
+        self.steps = get_sim_steps(time, unit)
+        self.unit = unit
+
+    @staticmethod
+    def _gcd_time_unit(*units):
+        unit_order = [
+            None,
+            'fs',
+            'ps',
+            'ns',
+            'us',
+            'ms',
+            'sec']
+        return unit_order[min(unit_order.index(u) for u in units)]
+
+    @property
+    def femtoseconds(self):
+        return get_time_from_sim_steps(self.steps, 'fs')
+
+    @property
+    def picoseconds(self):
+        return get_time_from_sim_steps(self.steps, 'ps')
+
+    @property
+    def nanoseconds(self):
+        return get_time_from_sim_steps(self.steps, 'ns')
+
+    @property
+    def microseconds(self):
+        return get_time_from_sim_steps(self.steps, 'us')
+
+    @property
+    def milliseconds(self):
+        return get_time_from_sim_steps(self.steps, 'ms')
+
+    @property
+    def seconds(self):
+        return get_time_from_sim_steps(self.steps, 'sec')
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self.steps == other.steps
+
+    def __add__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        result = SimTime(self.steps + other.steps)
+        result.unit = self._gcd_time_unit(self.unit, other.unit)
+        return result
+
+    def __sub__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        result = SimTime(self.steps - other.steps)
+        result.unit = self._gcd_time_unit(self.unit, other.unit)
+        return result
+
+    def __mul__(self, other):
+        if not isinstance(other, (int, float)):
```

Comment:
This precludes more exotic numbers like `fraction.Fraction`, `decimal.Decimal`, or `np.uint64`